src/app/layout/view-contact/view-contact.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Contact, ContactsService } from 'src/app/services/contacts.service';
+
+@Component({
+  selector: 'app-view-contact',
+  templateUrl: './view-contact.component.html',
+  styleUrls: ['./view-contact.component.css']
+})
+
+export class ViewContactComponent {
+ 
+  contact : Contact | null = null
+  constructor(private contactsService: ContactsService, private readonly route : ActivatedRoute ) {}
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe((params) => {
```

Comment:
Remember to store the subscription and unsubscribe in `onDestroy` otherwise you will have a memory leak